src/webhooks/twitchWebhookValidator.ts

```diff
@@ -0,0 +1,30 @@
+import { log, LogLevel } from '../common'
+import * as Crypto from 'crypto'
+
+export abstract class TwitchWebhookValidator {
+
+    public static validate(request, response, next) {
+
+        const givenSignature = request.headers['x-hub-signature'];
+      
+        if (!givenSignature) {
+            log(LogLevel.Error, `webhooks: validator - missing signature`)
+            return response.status(409).json({
+                error: 'Missing signature'
+            });
+        }
+
+        let digest = Crypto.createHmac('sha256',process.env.TWITCH_CLIENT_SECRET)
```

Comment:
We're using the Twitch client secret as the secret in the webhook. I'm wondering if that's a good (read: safe) idea.

The reason I think it's not is that when we register the webhook, we'll have to send the secret we want Twitch to use. I'm not feeling good about sending the Twitch client secret in that request since it could potentially be grabbed and used nefariously.

Should we generate a secret (maybe a GUID) to use? Send that GUID to Twitch and keep up with it on our end to use in validating these requests.

@parithon am I overthinking?